controllers/infra/api/v1/infra_types.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2022 labring.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"github.com/labring/sealos/pkg/types/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+/*
+apiVersion: infra.sealos.io/v1
+kind: Infra
+metadata:
+  name: aws-infra-demo
+  annotations:
+    infra.sealos.io/VpcID: v-xxxxx
+    infra.sealos.io/RouteID: v-xxxxx
+    infra.sealos.io/IgwID: v-xxxxxx
+    infra.sealos.io/EgwID: v-xxxxxx
+    infra.sealos.io/SubnetID: v-xxxxx
+    infra.sealos.io/SecurityGroupID: v-xxxxx
+    infra.sealos.io/EIPID: v-xxxxx
+spec:
+    regionIds: [cn-hangzhou, cn-shanghai]
+    zoneIds: [cn-hangzhou-a, cn-hangzhou-b]
+    ssh:
+       passwd: xxx
+       pk: /root/.ssh/id_rsa
+       port: 22
+       user: root
+   hosts:
+   - roles: [master, aaa, bbb] # required
+     count: 3 # Required
+     # key values resources.
+     resources:
+        cpu: 2
+        memory: 4
+        # other resources like GPU
+     # ENUM: amd64/arm64 (NOTE: the default value is amd64)
+     flavor: ecs.t5-lc1m2.large
+     arch: amd64
+     # ENUM: ubuntu:20.04, centos:7.2 and so on.
+     image: utuntu:20.04
+     disks:
+     - capacity: 50
+       # ENUM: system/data
+       type: system
+*/
+
+type Hosts struct {
+	Roles []string `json:"hosts,omitempty"`
+	Count int      `json:"count,omitempty"`
+	// key values resources.
+	// cpu: 2
+	// memory: 4
+	// other resources like GPU
+	Resources map[string]string `json:"resources,omitempty"`
```

Comment:
```suggestion
	Resources map[string]interface{} `json:"resources,omitempty"`
```
should resources support number|string| subarray?

Reply:
+1

Reply:
```
type Quantity struct {
	// i is the quantity in int64 scaled form, if d.Dec == nil
	i int64Amount
	// d is the quantity in inf.Dec form if d.Dec != nil
	d infDecAmount
	// s is the generated value of this quantity to avoid recalculation
	s string

	// Change Format at will. See the comment for Canonicalize for
	// more details.
	Format
}
```

Kuberentes resource using Quantity struct.